src/redux/missions/missions.js

```diff
@@ -0,0 +1,57 @@
+// Actions
+const START = 'space-ship/missions/START_LOADING_MISSIONS';
+const LOAD = 'space-ship/missions/LOAD_MISSIONS';
+const ADD_REMOVE_MEMBER = 'space-ship/missions/ADDREMOVE_MEMBER';
+
+const initialState = {
+  missions: [],
+  loading: true,
+  error: null,
+};
+  // Reducer
+export default function missionsReducer(state = initialState, action = {}) {
+  const { value } = action;
+  switch (action.type) {
+    case START:
+      return {
+        ...state,
+        loading: true,
+        error: null,
+      };
+    case LOAD:
+      return {
+        ...state,
+        loading: false,
+        error: value.error,
+        missions: value,
+      };
+    case ADD_REMOVE_MEMBER:
+      return {
+        ...state,
+        loading: false,
+        error: value.error,
+        missions: state.missions.map((mission) => {
+          const prevValue = mission.isMember;
+          if (mission.mission_id !== value) return mission;
+          return { ...mission, isMember: !prevValue };
+        }),
+      };
+
+    default:
+      return state;
+  }
+}
+
+// Action Creators
+
+export function startLoadingMessions(result) {
+  return { type: START, value: result };
+}
+
+export function LoadMessions(result) {
+  return { type: LOAD, value: result };
+}
+
+export function addremoveMembertoMission(result) {
+  return { type: ADD_REMOVE_MEMBER, value: result };
+}
```

Comment:
- [ ] Consider fixing the demonstrated issue below. When a user clicks the "Reserve rocket" button or "Joined Mission" button _[only if your team has 3 members]_, action needs to be dispatched to update the store. You need to get the ID of the reserved rocket and update the state. Remember you mustn't mutate the state. Instead, you need to return a new state object with all rockets, but the selected rocket will have an extra key `reserved` with its value set to `true`.

https://user-images.githubusercontent.com/52143815/193415679-cd260e96-519d-4f00-b91c-8eb1c5c4b554.mp4